### PR TITLE
Add support for versioning on index store

### PR DESCRIPTION
**Pull request by marclennox**
Add support for versioning (et al) to the individual document store method on index, similar to what is supported on bulk store.

### CODE REVIEW

**Comment by karmi** [comment]
Can you add unit tests in the https://github.com/karmi/tire/blob/master/test/unit/index_test.rb#L243 context?

**Comment by marclennox** [comment]
Will do, just need to figure out the version_type issue first.

**Comment by marclennox** [comment]
FYI, I've got versioning working in the PR both for bulk and individual updates but haven't resolved the testing failures or created new tests yet.

**Comment by marclennox** [comment]
Rebased and squashed my commits.  Any chance to get this merged?

**Comment by karmi** [comment]
Marc, I'm sorry it took so long to get to this. I see that the diff for the patch is really long, with many formatting changes apart from the code you added... Has your editor/IDE settings "improved" the formatting in this way or was it intentional? :)

File: lib/tire/index.rb [hunk 1]
```diff
@@ -172,23 +188,24 @@ def bulk(action, documents, options={})
...
 
-        header = { action.to_sym => { :_index => name, :_type => type, :_id => id } }
+        header = {action.to_sym => {:_index => name, :_type => type, :_id => id}}
 
         if document.respond_to?(:to_hash) && hash = document.to_hash
           meta = {}
-          meta[:_version]   = hash.delete(:_version)
-          meta[:_routing]   = hash.delete(:_routing)
+          meta[:_version] = hash.delete(:_version)
+          meta[:_version_type] = hash.delete(:_version_type)
```
Comment (karmi) [inline lib/tire/index.rb:202]: Is this the meat of the change?

**Comment by marclennox** [comment]
No there's more to it than that.  Let me resubmit and rebase without the formatting changes.

**Pull request closed without merging**